vms/avm/vm.go

```diff
@@ -387,6 +389,14 @@ func (vm *VM) LastAccepted(context.Context) (ids.ID, error) {
 	return vm.chainManager.LastAccepted(), nil
 }
 
+func (vm *VM) GetBlockIDAtHeight(_ context.Context, height uint64) (ids.ID, error) {
+	return vm.state.GetBlockIDAtHeight(height)
+}
+
+func (*VM) VerifyHeightIndex(context.Context) error {
```

Comment:
This is never called because it is wrapped by the HeightIndexedVM?

(if so, may make sense to clarify that)

Reply:
The vm itself implements HeightIndexedVM, it doesn't wrap it. This function is called [here](https://github.com/ava-labs/avalanchego/blob/master/vms/proposervm/vm.go#L384) when we're generating the proposervm height index.

Reason why it's `nil` is because the AVM has always had a height index so we don't need to verify it.

Reply:
How does it always have it if it is performed async?

Reply:
The AVM height index is not performed async. Since we linearized, each block was included in the height index on accept.

This is different than the P-chain where there is a period of time where both the inner vm and the proposervm height indexes are incomplete so we cannot simply return `nil` there.